### PR TITLE
Add ability to manually select content folder

### PR DESCRIPTION
**Pull request by ilcheese2**
### What is the new feature?
It allows you to select a content folder if one can't be found.
### Why should this be part of tModLoader?
It provides a fallback for when the folder can't be automatically detected. 
### Are there alternative designs?
It could replace the GOG portion entirely but that may be inconvenient for players. 

Continues off https://github.com/tModLoader/tModLoader/pull/3173.

### CODE REVIEW

File: patches/tModLoader/Terraria/nativefiledialog.cs.patch [hunk 1]
```diff
@@ -9,3 +9,12 @@
  		return result;
  	}
  
+@@ -107,7 +_,7 @@
+ 		IntPtr outPath2;
+ 		nfdresult_t result = INTERNAL_NFD_PickFolder(intPtr, out outPath2);
+ 		Marshal.FreeHGlobal((IntPtr)(void*)intPtr);
+-		outPath = UTF8_ToManaged(outPath2, freePtr: true);
++		outPath = UTF8_ToManaged(outPath2, freePtr: false);
```
Comment (Chicken-Bones) [inline patches/tModLoader/Terraria/nativefiledialog.cs.patch:17]: What's the reason for this change? Needs a comment at least

File: patches/tModLoader/Terraria/Main.TML.cs [hunk 2]
```diff
@@ -170,37 +171,62 @@ public partial class Main
...
+						vanillaContentFolder = Platform.IsOSX ? "../Terraria.app/Contents/Resources/Content" : "../Content"; // Nested Manual Install
+					}
 				}
-				Logging.tML.Info("Content folder of Terraria GOG Install Location assumed to be: " + Path.GetFullPath(vanillaContentFolder));
+
+				if (!IsValidContentFolder(vanillaContentFolder, out _)) {
+					nativefiledialog.NFD_PickFolder(null, out vanillaContentFolder);
+				}
+
+				File.WriteAllText("LaunchUtils/TerrariaFolderPath.txt", vanillaContentFolder);
```
Comment (Chicken-Bones) [inline patches/tModLoader/Terraria/Main.TML.cs:208]: Recommend moving this down below, as a 'common retry' code. Allows you to simplify some of the conditions here.

Also recommend only writing the `LaunchUtils/TerrariaFolderPath.txt` file if it's user selected.

Also better not to have user configuration items in the `LaunchUtils` folder. Just save this in the root directory and call it `ContentFolderPath.txt`

**Review by Chicken-Bones — CHANGES_REQUESTED** [review]
Overall, I'm not sure this feature is justified. `InstallVerifier` already has pathing checks for even allowing you to launch the game on GoG.

I think instead, we should replace all of the GoG pathing code to require the `Content` folder to be alongside the `vanillaExePath` from `InstallVerifier`

File: patches/tModLoader/Terraria/nativefiledialog.cs.patch (hunk 1 above)
Comment (ilcheese2) [reply patches/tModLoader/Terraria/nativefiledialog.cs.patch:17]: It's the same reason the other patch exists. I will add a comment.

**Comment by Chicken-Bones** [comment]
Closing as not worth it, can always be re-opened later
